### PR TITLE
Remove usage of Java assert in tests

### PR DESCRIPTION
**Pull request by jxnu-liguobin**
References #30220

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java [hunk 1]
```diff
@@ -57,7 +59,7 @@ public void demonstrateDeployment() {
         system.actorOf(
             Props.create(SampleActor.class).withDeploy(new Deploy(new RemoteScope(addr))));
     // #deploy
-    assert ref.path().address().equals(addr);
+    assertEquals(ref.path().address(), addr);
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:62]: the order for `assertEquals` is `expected, actual` - so perhaps they should be the other way around?

Comment (jxnu-liguobin) [reply akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java]: yes, i have updated it

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java [hunk 2]
```diff
@@ -4,24 +4,20 @@
...
 import akka.testkit.AkkaJUnitActorSystemResource;
+import akka.testkit.AkkaSpec;
 import jdocs.AbstractJavaTest;
 import org.junit.ClassRule;
 import org.junit.Test;
 
 import com.typesafe.config.ConfigFactory;
 
 // #import
-import akka.actor.ActorRef;
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:15]: These imports are useful to include in the documentation snippets generated from this source file, could you put them back?

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java [hunk 3]
```diff
@@ -34,7 +30,14 @@ public Receive createReceive() {
...
   public static AkkaJUnitActorSystemResource actorSystemResource =
-      new AkkaJUnitActorSystemResource("RemoteDeploymentDocTest");
+      new AkkaJUnitActorSystemResource(
+          "RemoteDeploymentDocTest",
+          ConfigFactory.parseString(
+                  "   akka.actor.provider = remote\n"
+                      + "    akka.remote.classic.netty.tcp.port = 0\n"
+                      + "    akka.remote.artery.canonical.port = 0\n"
+                      + "    akka.remote.use-unsafe-remote-features-outside-cluster = on")
+              .withFallback(AkkaSpec.testConf()));
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:40]: This change doesn't seem to be related to this PR?

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java [hunk 4]
```diff
@@ -49,15 +52,16 @@ void makeAddress() {
...
     // #make-address
     // #deploy
-    ActorRef ref =
-        system.actorOf(
-            Props.create(SampleActor.class).withDeploy(new Deploy(new RemoteScope(addr))));
+    Props props = Props.create(SampleActor.class).withDeploy(new Deploy(new RemoteScope(addr)));
+    ActorRef ref = system.actorOf(props);
     // #deploy
-    assert ref.path().address().equals(addr);
+    assertEquals(addr, ref.path().address());
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:64]: :+1:

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java [hunk 5]
```diff
@@ -49,15 +52,16 @@ void makeAddress() {
   @Test
   public void demonstrateDeployment() {
     // #make-address
-    Address addr = new Address("akka", "sys", "host", 1234);
-    addr = AddressFromURIString.parse("akka://sys@host:1234"); // the same
+    Address addr = new Address("akka", "sys", "localhost", 1234);
+    addr =
+        AddressFromURIString.parse(
+            String.format("%s://%s@localhost:%d", "akka", "sys", 1234)); // the same
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:58]: I don't see how this is better?

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java (hunk 3 above)
Comment (jxnu-liguobin) [reply akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:40]: If not, assert will be false.

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java (hunk 2 above)
Comment (jxnu-liguobin) [reply akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:15]: ok

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java (hunk 3 above)
Comment (jxnu-liguobin) [reply akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:40]: I think we need a RemoteProvider to create a remote actorRef, This is the same as `RemoteDeploymentDocSpec.scala`

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java (hunk 5 above)
Comment (jxnu-liguobin) [reply akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:58]: Yes, it's useless.

File: akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java (hunk 3 above)
Comment (raboof) [reply akka-docs/src/test/java/jdocs/remoting/RemoteDeploymentDocTest.java:47]: aah, gotcha, thanks for the clarification

**Review by raboof — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM
